### DISCUSSION TITLE
[유니드컴즈] 커머스 API 센터 > 솔루션 등록하기(수정하기) 에 대해 문의드립니다.

### DISCUSSION THREAD

**Discussion by Guwon** [commerce-api-naver/commerce-solutions#33] [묻고 답하기]
안녕하세요.
유니드컴즈 김훈입니다.

현재 이미 등록되어 있는 솔루션에 대해 이벤트 훅 및 요금제 영역에 대한 수정을 진행중인데요
이미 등록되어 있는 솔루션에 대한 수정을 진행하기 위해서는 `API호출 IP`영역에 새로 기입을 해야하는데
관련해서 궁금한점이 생겨 문의드립니다.

1. `API호출 IP`가 적용되는 범위
   커머스 솔루션 권한에 해당하는 API의 경우 솔루션이 구독(이용중) 상태가 되기전에 호출하는 경우가 있는데요
   해당 권한에 속하는 API도 `API호출 IP`에 영향을 받는걸까요? 

2. 지금 대부분 솔루션이 `API호출 IP`가 입력되지 않은 상태인데 
    지금 상태와 동일하게 유지하면서 `API호출 IP`를 입력하기 위해선 모든 IP에 대한 접근을 허용 하는 '0.0.0.0' 을 기입하면 될까요?
    AWS의 인바운드 규칙과 동일한 개념으로 이해하고 있습니다.

3. 특정 IP를 입력할 경우 솔루션을 수정하는 즉시 적용되어 
    해당 IP를 제외한 나머지 IP는 바로 차단이 적용되는 걸까요?

![image](https://github.com/user-attachments/assets/cace4c2b-8d93-4df1-83fd-9822ed6c7d81)

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

### API호출 IP가 적용되는 범위
대상하는 애플리케이션의 모든 API 호출에 반영됩니다.
통상적인 솔루션의 구독 처리 과정에서 커머스API 호출은 솔루션 시스템이 진행하는 경우가 대부분이기에 사용자의 솔루션 구독이 완료되기 이전이더라도 유니드컴즈 시스템에서 API를 호출한다면 사용자의 솔루션 구독 상태에는 영향이 없습니다.

커머스솔루션마켓에서의 사용자 구독과 '사용 후 승인'형 솔루션에만 해당하는 '판매자 커머스ID 인증(프론트-엔드)'을 제외한 대상 애플리케이션에 적용되는 커머스API 호출건에 IP가 적용된다고 이해해주시면 됩니다.

### IP 적용 규칙
AWS의 인바운드 규칙과는 다른 개념입니다.
모든 IP에 대한 허용은 불가능하며 대상 애플리케이션으로의 모든 커머스API 호출에 관련한 IP Whitelist라고 이해해주시면 될 것 같습니다.
~~(등록되지 않은 IP로는 API 호출 불가)~~

### 적용 후 즉시 차단 여부
~~네 맞습니다.~~
~~상황에 따라 실제 시스템 적용까지 약간의 시간차가 있을 수 있으나, 통상적인 케이스에서 설정 저장 즉시 적용으로 이해해주시면 됩니다.~~

> [!NOTE]
> (10/17 14:54 정정) 2024년 4분기 중 솔루션의 안전관리와 관련한 일련의 기능 업데이트를 준비하고 있습니다.
> 해당 시점 이후 등록된 IP 이외 다른 IP에서의 커머스API 호출이 불가능할 수 있으며 상세 내용은 차후 별도로 공지할 예정입니다.

**Reply by Guwon** [reply]
@commerce-api-naver 
안녕하세요. 
유니드컴즈 김훈입니다.

회신 감사드립니다.

지금 현재는 대부분의 상품에 IP 등록이 안되어있는데
현재는 모든 접근에 대해 허용이 되어있다고 이해할 수 있을까요?

**Reply by commerce-api-naver** [reply]
@Guwon 님, 커머스API 이민석입니다.

앞선 [답변](https://github.com/commerce-api-naver/commerce-solutions/discussions/33#discussioncomment-10966882)에서 몇가지 정정해드릴 내용이 있어 기존 답변을 수정하고 새로운 답변으로 알려드립니다.

1. 2024년 4분기 중 솔루션의 안전관리 기능 업데이트(가칭)를 준비하고 있으며, 이 과정에서 등록된 IP 이외 다른 IP로의 커머스API 호출이 불가능할 수 있습니다. (상세 내용 별도 공지 예정)
3. 시범 연동 과정에서 오픈한 솔루션들은 IP 등록이 되어있지 않은 상태로 솔루션 안전관리 기능 업데이트 과정에서 관련한 안내가 발송될 수 있습니다. (사전 입력을 권해드립니다.)

**Reply by Guwon** [reply]
@commerce-api-naver 
안녕하세요.
유니드컴즈 김훈입니다.

그러면 현재는 등록된 IP 외에도 호출이 가능하고
2024년 4분기 안전관리 기능이 업데이트 될 경우에는
등록된 IP 외에는 호출이 불가능할 수 있다는 말씀으로 이해했는데 
제가 이해한 내용이 맞는지 확인 부탁드립니다.

**Reply by commerce-api-naver** [reply]
@Guwon 님, 커머스API 이민석입니다.

이해하신 내용이 맞습니다.
관련된 내용은 저희 내부적으로 좀 더 다듬어 공지사항 등으로 다시 한 번 알려드리도록 하겠습니다.

**Reply by Guwon** [reply]
@commerce-api-naver 
안녕하세요.
유니드컴즈 김훈입니다.

지금 커머스 API 센터를 통해 `IP 등록` 및 `신규 이벤트 훅` 설정중에 있는데요.

솔루션 수정을 완료 할 경우 
커머스 솔루션 마켓에서 상품 신청시 JWT 토큰이 파라미터에 달려서 오고 있습니다.
해당 토큰은 `JWT 바로승인형` 솔루션에서 제공되는 토큰으로 알고 있는데
해당 상품은 `심사후 승인형` 임에도 불구하고 토큰이 함께 들어오고 있어서 문의드립니다.

<솔루션 수정되어 신청시 이슈가 발생하는 솔루션 목록>
메타 구매전환 추적 AI 매니저, 
네이버 쇼핑검색광고 AI 매니저, 
리뷰 노출 AI 매니저 ,
인스타그램 상품관리 AI 매니저,
메타 광고소재 제작 AI 매니저 

![image](https://github.com/user-attachments/assets/7190722e-a7a8-4e66-95ed-34a23926715c)

**Reply by commerce-api-naver** [reply]
@Guwon 님 안녕하세요. 커머스API 이민석입니다.

커머스API센터를 통한 솔루션 관리 기능을 제공하며 모든 솔루션은 '바로 사용'형 이용(JWT 파라미터 제공)이 기본이 됩니다.
이 중 원하는 솔루션에 한하여만 추가적인 옵션 형태로 '사용 후 승인'형 이용을 제공합니다.

따라서 커머스API센터에서 솔루션 정보를 수정하신 경우 기본 제공 정책으로 설정 내용이 업데이트 된 것으로 보입니다.
혹시 변경된 정책이 유니드컴즈의 솔루션 운용에 영향이 있다면 알려주실 수 있을까요?

**Reply by Guwon** [reply]
@commerce-api-naver 
안녕하세요. 
유니드컴즈 김훈입니다.

솔루션에 따라 토큰 해석 작업이 달라서 이슈가 있었는데 
현재는 대응하도록 수정되었습니다. 확인해주셔서 감사합니다.

**Comment by commerce-api-naver** [comment]
@Guwon 님 안녕하세요. 커머스API 이민석입니다.

문의주신 건과는 별건으로 지난 10/14 시범 연동 개발사 공간으로 요청주신 "메타 구매전환 추적 AI 매니저" 솔루션의 요금제 추가건과 관련하여 현재 요청주신 변경 요금제로 노출되고 있으나 요금제 연동이 안되어 오류가 발생하고 있습니다.
추가드린 요금제의 Plan ID는 해당 이슈 코멘트로 전달드렸는데요, 확인 가능하실까요?
- https://github.com/commerce-api-naver/solutions-early/issues/568

**Reply by Guwon** [reply]
@commerce-api-naver 
안녕하세요 유니드컴즈 유동진입니다. 요금제 연동 오류에 대해 상세하게 전달해 주실 수 있을까요?

저희 쪽 연동 요금제 응답은 기존 프로모션과 신규 프로모션 사용자를 구분하여 응답하고 있습니다.
첫 구독자와 오류 발생 시에는 기존 베이직 요금제(4lBatgKlmAh7HQ9BYwFSvC)로 응답하고 있는데 이 부분이 문제가 될까요?

**Reply by commerce-api-naver** [reply]
@Guwon 님 안녕하세요. 커머스API 이민석입니다.

기존 요청의 결과로 총 3벌의 요금제가 추가되었습니다. (베이직 / 스탠다드 / 프리미엄)
베이직 요금제 역시 내부적으로는 요청주신 일자 전과 후가 다른 요금제ID로 구분되어 베이직 요금제에 대한 요금제ID 응답을 신규 ID로 기대하고 있습니다.

이 부분은 유니드컴즈 측에서 신규 베이직 요금제의 요금제ID로 교체 응답이 가능하실지,
아니면 저희가 베이직 요금제에 한하여 기존 요금제 정보로 복원하면 될지 확인 부탁드리겠습니다.

**Reply by Guwon** [reply]
@commerce-api-naver 
확인 감사합니다.

해당 상품의 평생 할인 프로모션은 종료되더라도 기존 구독자들의 요금제는 유지되어야 합니다.
이에 따라 저희는 사용자 구독 상태, 판매자 등급, 프로모션 여부를 기준으로 적절한 요금제를 응답하고 있습니다.

커머스 측에서 구독(재구독) 또는 정기 결제 시 연동 요금제를 호출했을 때, 저희가 응답한 요금제(솔루션의 전체 요금제 중)로 처리가 가능하실지 확인 부탁드립니다.

**Reply by Guwon** [reply]
추가로 해당 이슈가 즉시 해결하기 어렵다면 기존 요금제 노출을 고려하고 있습니다. 이때 연동 요금제 호출 이슈가 있을지 함께 확인 부탁드립니다.

**Reply by commerce-api-naver** [reply]
@Guwon 님, 커머스API 이민석입니다.

이 문제는 저희가 별도 채널로 다시 연락드리겠습니다.

**Reply by Guwon** [reply]
넵 감사합니다.
